code/java/Example.java

```diff
@@ -1,35 +1,32 @@
-// Add your the driver dependency to your pom.xml build.gradle etc.
-// Java Driver Dependency: http://search.maven.org/#artifactdetails|org.neo4j.driver|neo4j-java-driver|4.0.1|jar
+// Use Java 17 for this example.
+// Add the driver dependency to your pom.xml build.gradle etc.
+// Java Driver Dependency:  http://search.maven.org/#artifactdetails|org.neo4j.driver|neo4j-java-driver|4.4.9|jar
 // Reactive Streams http://search.maven.org/#artifactdetails|org.reactivestreams|reactive-streams|1.0.3|jar
 // download jars into current directory
 // java -cp "*" Example.java
 
-import org.neo4j.driver.*;
-import static org.neo4j.driver.Values.parameters;
+import org.neo4j.driver.AuthTokens;
+import org.neo4j.driver.GraphDatabase;
+import org.neo4j.driver.Query;
+import org.neo4j.driver.SessionConfig;
 
-public class Example {
-
-  public static void main(String...args) {
-
-    Driver driver = GraphDatabase.driver("neo4j+s://demo.neo4jlabs.com:7687",
-              AuthTokens.basic("mUser","s3cr3t"));
-
-    try (Session session = driver.session(SessionConfig.forDatabase("movies"))) {
-
-      String cypherQuery =
-        "MATCH (m:Movie {title:$movieTitle})<-[:ACTED_IN]-(a:Person) RETURN a.name as actorName";
+import java.util.Map;
+import java.util.stream.Collectors;
 
-      var result = session.readTransaction(
-        tx -> tx.run(cypherQuery, 
-                parameters("movieTitle","The Matrix"))
-            .list());
-
-      for (Record record : result) {
-        System.out.println(record.get("actorName").asString());
-      }
+public class Example {
+    public static void main(String[] args) {
+        try (var driver = GraphDatabase.driver("neo4j+s://demo.neo4jlabs.com:7687", AuthTokens.basic("mUser", "s3cr3t"));
+             var session = driver.session(SessionConfig.forDatabase("movies"))) {
+            var query = new Query("""
```

Comment:
can we not use a Query Object?

Reply:
tying query string and parameters into one is a really uncommon way as parameters are dynamic and query string is not also it's not idiomatic for java database access

Reply:
> can we not use a Query Object?

Could you clarify what you mean please? It seems to use the `Query` object.

Reply:
I think he wants to ask if it's possible to avoid using a Query object because

> tying query string and parameters into one is a really uncommon way as parameters are dynamic and query string is not also it's not idiomatic for java database access